File: Python/1.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_table = {}
        for i, v in enumerate(nums):
            hash_table[v] = i

        for i, v in enumerate(nums):
            if (target - v) in hash_table and hash_table[target-v] != i:
                return [i, hash_table[target-v]]

File: Golang/1/1.go
package twoSum

func twoSum(nums []int, target int) []int {
	hashtable := make(map[int]int)
	var result []int
	for index, num := range nums {
		hashtable[target-num] = index
	}
	for index, num := range nums {
		if val, ok := hashtable[num]; ok && val != index {
			result = []int{val, index}
		}
	}
	return result
}
